### PR TITLE
[docs] Update contribution guidelines

### PR DESCRIPTION
**Pull request by suhanime**
Added PR workflow

### CODE REVIEW

File: CONTRIBUTION.md [hunk 1]
```diff
@@ -64,3 +64,16 @@ Bug: T42
...
+- Before submitting a PR
+  - Format your code and fix all the spelling/grammatical mistakes.
+  - Limit the column length to 120 characters.
+  - Error messages should not be more than 2 lines long.
+- Following are the things you should keep in mind once you open a PR:
+  - Wait for comments from at least 2 reviewers before pushing changes. Open comments in the meanwhile can be worked on locally.
+  - PR comments should be addressed in new commits. Before final approval, they have to be squashed.
+  - Maintain changes to multiple commits. (see below)
+
+![Sample PR life-cycle](/images/PR-workflow.jpg)
```
Comment (ravisantoshgudimetla) [inline CONTRIBUTION.md]: Perhaps some text explaining the diagram would help.

File: CONTRIBUTION.md [hunk 2]
```diff
@@ -64,3 +64,16 @@ Bug: T42
 The first line is the subject and should be no longer than 50 characters, the second line is always blank, and other
 lines should be wrapped at 80 characters. This allows the message to be easier to read on GitHub as well as in various
 git tools.
+
+### PR workflow
+
+- Before submitting a PR
+  - Format your code and fix all the spelling/grammatical mistakes.
+  - Limit the column length to 120 characters.
+  - Error messages should not be more than 2 lines long.
```
Comment (ravisantoshgudimetla) [inline CONTRIBUTION.md]: I think what @aravindhp meant is single line here. The error messages should not be more than 1 line but I'll let him speak for it

Comment (aravindhp) [reply CONTRIBUTION.md]: Yes, I meant single line here.

File: CONTRIBUTION.md (hunk 1 above)
Comment (aravindhp) [reply CONTRIBUTION.md]: The diagram on its own will be hard to interpret. I would let @gmarkley-VI come up with the diagram and you can add it in a follow up PR. For now you could a few sentences explaining what maintaning changes to multiple commits mean. How about `If your PR has multiple commits, changes requested should eventually be squashed into the original commit where the change was requested before the final merge.`?

File: CONTRIBUTION.md [hunk 3]
```diff
@@ -64,3 +64,16 @@ Bug: T42
...
+- Following are the things you should keep in mind once you open a PR:
+  - Wait for comments from at least 2 reviewers before pushing changes. Open comments in the meanwhile can be worked on locally.
```
Comment (aravindhp) [inline CONTRIBUTION.md]: Not sure if this within 120 col. I too missed ensuring that in line 33. You are welcome to fix that here if you would like.

File: CONTRIBUTION.md (hunk 1 above)
Comment (suhanime) [reply CONTRIBUTION.md]: Sounds good. I'll try to use this, or come up with a couple of sentences to explain well in the next commit.

File: CONTRIBUTION.md (hunk 3 above)
Comment (suhanime) [reply CONTRIBUTION.md]: Will do!

File: CONTRIBUTION.md [hunk 4]
```diff
@@ -64,3 +64,19 @@ Bug: T42
 The first line is the subject and should be no longer than 50 characters, the second line is always blank, and other
 lines should be wrapped at 80 characters. This allows the message to be easier to read on GitHub as well as in various
 git tools.
+
+### PR workflow
+
+- Before submitting a PR
+  - Format your code and fix all the spelling/grammatical mistakes.
+  - Limit the column length to 120 characters.
```
Comment (ravisantoshgudimetla) [inline CONTRIBUTION.md]: Can you be explicit on `column length` of what? Code or comments within code or the commit message?

File: CONTRIBUTION.md [hunk 5]
```diff
@@ -64,3 +64,19 @@ Bug: T42
...
+  - Error messages should be limited to a single line.
```
Comment (ravisantoshgudimetla) [inline CONTRIBUTION.md]: `Error messages within code should be limited to?` perhaps??

File: CONTRIBUTION.md (hunk 4 above)
Comment (suhanime) [reply CONTRIBUTION.md]: Commit message format highlights the column length for a commit message, so will change this line to mention code as well as comments within code.

**Comment by ravisantoshgudimetla** [comment]
@suhanime - Can you please squash the commits?

**Comment by ravisantoshgudimetla** [comment]
/lgtm
Thanks @suhanime for the PR

**Pull request merged**